lib/rules/prefer-lookaround.ts

```diff
@@ -172,6 +185,205 @@ function getSideEffectsWhenReplacingCapturingGroup(
+    // e.g.
+    // /foo(bar)$/ -> $
+    // /foo(bar)\b/ -> \b
+    // /foo(bar)(?:\b|$)/ -> (?:\b|$)
+    // /foo(bar)/ -> null
+    trailingAssertion: TrailingAssertion | null
+    // The pattern used when replacing lookahead assertions.
+    replacedAssertion: string
+    range: PatternRange
+}
+type ParsedElements = {
+    // All elements
+    elements: readonly Element[]
+    start: ParsedStartPattern | null
+    end: ParsedEndPattern | null
+}
+
+/**
+ * Parse the elements of the pattern.
+ */
+function parsePatternElements(node: Pattern): ParsedElements | null {
+    if (
+        node.alternatives.length > 1 ||
+        node.alternatives[0].elements.length < 2
+    ) {
+        return null
+    }
+    const elements = node.alternatives[0].elements
+    let start: ParsedStartPattern | null = null
+    let leadingAssertion: LeadingAssertion | null = null
+    let startCandidate = elements[0]
+    if (isLeadingAssertion(startCandidate)) {
+        leadingAssertion = startCandidate
+        startCandidate = elements[1]
+    }
+    if (
+        startCandidate?.type === "CapturingGroup" &&
+        !isZeroLength(startCandidate)
+    ) {
+        const capturingGroup = startCandidate
+        start = {
+            leadingAssertion,
+            capturingGroup,
+            replacedAssertion: `(?<=${assertionToLookaroundAssertionPatternText(
+                leadingAssertion,
+            )}${capturingGroup.alternatives.map((a) => a.raw).join("|")})`,
+            range: leadingAssertion
+                ? {
+                      start: leadingAssertion.start,
+                      end: capturingGroup.end,
+                  }
+                : capturingGroup,
+        }
+    }
+
+    let end: ParsedEndPattern | null = null
+    let trailingAssertion: TrailingAssertion | null = null
+    let endCandidate = elements[elements.length - 1]
+    if (isTrailingAssertion(endCandidate)) {
+        trailingAssertion = endCandidate
+        endCandidate = elements[elements.length - 2]
+    }
+    if (
+        endCandidate?.type === "CapturingGroup" &&
+        !isZeroLength(endCandidate)
+    ) {
+        const capturingGroup = endCandidate
+        end = {
+            capturingGroup,
+            trailingAssertion,
+            replacedAssertion: `(?=${capturingGroup.alternatives
+                .map((a) => a.raw)
+                .join("|")}${assertionToLookaroundAssertionPatternText(
+                trailingAssertion,
+            )})`,
+            range: trailingAssertion
+                ? {
+                      start: capturingGroup.start,
+                      end: trailingAssertion.end,
+                  }
+                : capturingGroup,
+        }
+    }
+    if (!start && !end) {
+        return null
+    }
+
+    return {
+        elements,
+        start,
+        end,
+    }
+
+    /** Convert leading/trailing assertion to lookaround assertion pattern text. */
+    function assertionToLookaroundAssertionPatternText(
```

Comment:
This function has to make sure that the lookaheads/lookbehinds it unwraps look into the same direction as their parent. E.g. `foo(bar)(?<=z\w+)` -> `foo(barz\w+)` is wrong, it has to be `foo(bar(?<=z\w+))`.

Also, should this rule even unwrap lookarounds? Transforming `(?=foo(?=bar))` -> `(?=foobar)` seems like something a dedicated rule should do. Maybe [`no-trivially-nested-assertion`](https://ota-meshi.github.io/eslint-plugin-regexp/rules/no-trivially-nested-assertion.html) could do it?

Reply:
Oh, I didn't see #481. That also works for me :)

Reply:
> Also, should this rule even unwrap lookarounds?

In my opinion, we need to unwrap the simplest lookarounds. prefer-lookaround rule generate lookaround assertions with autofix.
I think the new report after autofix should combine the assertions made by the rule and autofix them.

```js
var str = 'JavaScript'.replace(/Java(S)(c)(r)(i)(p)(t)/g, 'Type$1$2$3$4$5$6')
// ↓
var str = 'JavaScript'.replace(/Java(S)(c)(r)(i)(p)(?=t)/g, 'Type$1$2$3$4$5')
// ↓
var str = 'JavaScript'.replace(/Java(S)(c)(r)(i)(?=pt)/g, 'Type$1$2$3$4')
// ↓
var str = 'JavaScript'.replace(/Java(S)(c)(r)(?>=ipt)/g, 'Type$1$2$3')
// ↓
// ...
// ↓
var str = 'JavaScript'.replace(/Java(?>=Script)/g, 'Type')
```

Reply:
Alright, then let's unwrap simple lookarounds.